src/api/src/resource/mod.rs

```diff
@@ -35,6 +35,18 @@ pub struct ResourceDesc {
     pub resource_tag: String,
 }
 
+impl ResourceDesc {
+    pub fn check_legality(&self) -> Result<()> {
```

Comment:
```suggestion
    pub fn is_valid(&self) -> Bool {
```

A terminology nit: legality refers to the actual law (as in judges and lawyers :) ). And I think returning a boolean seems more natural?